### DISCUSSION TITLE
[Feature suggestion] make Todo list available to sync with Google tasks (and maybe other services too)

### DISCUSSION THREAD

**Discussion by Alpaca131** [zepp-health/zeppos-forum#3] [Developer Feedback]
I feel Todo list very useful, but it's inconvenient that users need to launch Zepp app and tap some buttons just to edit the list. 
It's so convenient if users can connect some services like Google Tasks, Microsoft Todo, **so users can edit and notified same content on Mobile, Smart speaker, Desktop, and other devices** too.

**Comment by duanxiong** [comment]
@Alpaca131 , how about sync your tasks to the calendar (with the calendar setting in Zepp App on the mobile phone), the calendar can remind you with backgroud service which is not supported in Todo list and other JS apps.

**Reply by Alpaca131** [reply]
Unfortunately I've tried that but it didn't work.
It seems that Google Tasks do add Todo on Google Calendar app, but not on device's calendar. 
![Screenshot_20221109-104350_Zepp.png](https://user-images.githubusercontent.com/66709717/200730215-d4abd111-2f25-4ffd-8bfa-1552208b3aba.png)
![Screenshot_20221109-104339_Calendar.png](https://user-images.githubusercontent.com/66709717/200730237-daf0e362-2efc-4880-b5b6-648783f6bc2d.png)

**Reply by duanxiong** [reply]
Zepp App need to sync your tasks from the mobile phone's official calendar to the device, maybe you can try to sync the tasks between Google Calendar and your mobile phone's official calendar firstly?
![img_v2_091c5809-7323-4db4-8b4c-b5d450f8789g](https://user-images.githubusercontent.com/78085548/200749352-d181c396-9898-41a5-b323-f1fe24aad918.jpg)

**Reply by tumbleweed05** [reply]
The System Calendar Sync is incredibly flaky, not syncing all events and not syncing new events in a timely fashion. Surprised at this behavior.

**Comment by MyoungXUE** [comment]
I created a new category and moved the discussion to the developer feedback category. A colleague with extensive experience in open web interfaces will answer this question.

**Comment by fay20fay** [comment]
> I feel Todo list very useful, but it's inconvenient that users need to launch Zepp app and tap some buttons just to edit the list. It's so convenient if users can connect some services like Google Tasks, Microsoft Todo, **so users can edit and notified same content on Mobile, Smart speaker, Desktop, and other devices** too.

If Google Tasks, Microsoft Todo, these three-party services support Oauth authorization and have the relevant API to obtain user data, then users can directly edit the list on other devices such as cell phones, computers, etc. without having to click buttons in zepp to operate.
